requirements-tests.txt

```diff
@@ -1,25 +1,41 @@
-attrs>=20.2.0
-bcrypt>=3.1.6
-capturer >= 2.2
--r requirements-coverage.txt
-cryptography>= 2.3
-mock >= 1.0.1
-pyserial-asyncio>=0.4.0;python_version>="3.4"
-pep8>=1.7.0
-pyasn1>=0.4.2
-pyserial>=3.4
-pytest-cov>=2.5.1
+
+# -------------------------------------------------------------------
+# Python packages needed to run tests.
+# -------------------------------------------------------------------
+# Python packages required to run tests including:
+# - pylint
+# - codespell
+# - bandit
+# - coverage
+# - flake8
+# Which are all part of a normal CI run.
+#
+# REMARK: if you have commented out packages in requirements.txt it will fail.
+-r requirements.txt
+
+
+# -------------------------------------------------------------------
+# Python packages needed to run tests
+# -------------------------------------------------------------------
 pytest>=3.5.0
-redis>=2.10.6
-sqlalchemy>=1.1.15
-#wsgiref>=0.1.2
-verboselogs >= 1.5
-tornado==4.5.3
-# using platform_python_implementation rather than
-# implementation_name for Python 2 support
-Twisted[conch,serial]>=20.3.0; platform_python_implementation != "PyPy" or sys_platform != "win32"
-# pywin32 isn't supported on pypy
-# https://github.com/mhammond/pywin32/issues/1289
-Twisted[conch]>=20.3.0; platform_python_implementation == "PyPy" and sys_platform == "win32"
-zope.interface>=4.4.0
+pytest-cov>=2.5.1
+pep8>=1.7.0
 asynctest>=0.10.0
```

Comment:
I remember there was an issue with `asynctest` for some versions of pytest. Will have to check that out.

Reply:
ok

Reply:
Refer https://github.com/riptideio/pymodbus/issues/681

Reply:
In the actual versions of python, unittest support async stuff.
I would either recommened to switch to [builtin stuff](https://docs.python.org/3/library/unittest.html?highlight=unittest#unittest.IsolatedAsyncioTestCase) or as an alternative use the pytest plugin "[pytest-asyncio](https://github.com/pytest-dev/pytest-asyncio)".

['asynctest'  seems to be abandoned (last commit 2019)](https://github.com/Martiusweb/asynctest)

Reply:
That is a good hint !!!

I will take a look later and see how easy it is to remove, stay tuned.

Reply:
I think I am able to remove asynctest, I will do that before this one gets merged.

@dhoomakethu I removed package "attrs", because I could not find any references, but I just saw a commit in your "black" branch that upgrade the version, have I missed something ?

Reply:
I do not remember using `attrs` any where explicitly, I believe it came in as part of the `black` package dependency.

Reply:
thanks, yes it is a requisite for black, but we should not list the dependencies for other packages.